Vizzuality/marxan-cloud
# Export a dump from the dbs and generate a small data extraction centered in the okavango

## Export a dump from the DBs and generate a small data extraction centered in the Okavango.

### Overview

With this pr we close the DB management circle and test data generation; 
This adds a method to generate a dump of the DBs, upload them to azure, and repopulate a clean slate from it.
This also adds a pipe to generate a small version of the data stored in a full DBs, based on the Okavango, and generates a clean test SQL file that can be used to populate the test DBs.

### Testing instructions

You need to have the API running, and a full DB setup (if you don't have a full DB, run the below commands and grab a cup of tea, the normal process should take between 30 and 45 min to fulfill):
```bash
make start-api
make seed-api-with-test-data && make seed-geodb-data
```
These commands will generate a full dump of your DBs in `data/data/processed/db_dumps/`
```bash
make dump-geodb-data && make dump-api-data
```
You need to have the dumps ready in `data/data/processed/db_dumps/`, have `az` cli in your local, and being log in
```bash
make upload-dump-data
```
In order to restore the DB you can do:
```bash
make restore-dumps
```

And now that we have a full DB fresh and restored you can if you want, extract and generate new testing data:
Requirements: `make notebooks`
```bash
make generate-geo-test-data
```
or if you want to populate db with testing data just do:

```bash
make seed-dbs
```
